internal/datastore/crdb/tx.go

```diff
@@ -102,24 +74,19 @@ func executeWithResets(ctx context.Context, conn conn, txOptions pgx.TxOptions,
 	}()
```

Comment:
can't comment on the defer block directly, but this would unwind some of the nesting:
```go
	defer func() {
		if tx == nil {
                    return
                }
                // the rest
         }
```